packs/hpe-icsp/actions/lib/icsp.py

```diff
@@ -63,14 +63,31 @@ def get_sessionid(self):
 
         # added here due to the requirement of the session id
         self.base_headers = {'Auth': self.icsp_sessionid,
-                   'X-Api-Version': self.icsp_apiv}
+                             'X-Api-Version': self.icsp_apiv}
 
         return results["sessionID"]
 
     def extract_id(self, joburi):
         jobid = str(joburi)
         return int(jobid.split("/")[-1])
 
+    def get_mids(self, ids, idtype):
+        endpoint = "/rest/os-deployment-servers"
+        getresults = self.icsp_get(endpoint)
+        servers = getresults["members"]
+        mids = []
+        for id in ids:
+            for server in servers:
+                if idtype == 'serialnumber':
+                    if ((server["serialNumber"] == id) and
```

Comment:
i did used to have just one for loop (server in servers) although because of the way this function is called i need to return mid values in the same order as the ids they relate to in the parameter. using server in servers means that the mids are passed back based on the order the icsp server returns them not what the person inputs. Will see if i can think of a better cleaner way.